projects/testing-library/src/lib/models.ts

```diff
@@ -94,7 +94,11 @@ export type AliasedInputs = Record<string, AliasedInput<unknown>>;
 
 export type ComponentInput<T> =
   | {
-      [P in keyof T]?: T[P] extends Signal<infer U> ? U : T[P];
+      [P in keyof T]?: T[P] extends InputSignalWithTransform<any, infer U>
```

Comment:
I do think it's the case, but can you confirm that InputSignalWithTransform is also exported in Angular 17? (otherwise this is a breaking change)

Reply:
It seems to be.

I don't know how you would like me to confirm it but this is what I have done : 

1. Created a project under v17 with `npx @angular/cli@17 new angular-test`
2. Added a signal input with transform function and checked that I could explicitly type it 

```
  title: InputSignalWithTransform<number, string> = input.required<
    number,
    string
  >({ transform: numberAttribute });
```
3. Checked imports :
```
import {
  Component,
  input,
  InputSignalWithTransform,
  numberAttribute,
} from '@angular/core';
```

Does that look good to you ?

Reply:
Yea, I verified this as well like this after checking Angular's source code for v17.
Looks good, thanks for the quick PR and updates!